drivers/backup/backup.go

```diff
@@ -98,6 +99,11 @@ type Backup interface {
 
 	// DeleteBackup deletes backup
 	DeleteBackup(req *api.BackupDeleteRequest) (*api.BackupDeleteResponse, error)
+
+	// WaitForBackupCompletion waits for backup to complete successfully
+	// or till timeout is reached. API should poll every `timeBeforeRetry`
+	WaitForBackupCompletion(timeout time.Duration,
```

Comment:
Maybe lets put request first and timeouts and reties in the end of argument list?